### ISSUE TITLE
Rename BitVec to Vec

### ISSUE THREAD

**Issue by satnam6502** [project-oak/oak-hardware#276]
In the declaration:
```coq
Inductive Kind : Type :=
  | Void : Kind                    (* An empty type *)
  | Bit : Kind                     (* A single wire *)
  | BitVec : Kind -> nat -> Kind   (* Vectors, possibly nested *)
  | ExternalType : string -> Kind. (* An uninterpreted type *)
```
the constructor `BitVec` was so named because originally it was a flat bit-vector with no nesting. Now it can be a nested vector of vectors. Rename it to `Vec` to avoid confusion.

**Comment by blaxill** [comment]
Can it contain `ExternalType`/`Void`?

**Comment by satnam6502** [comment]
Do you mean should it contain `ExternalType/Void`?

**Comment by satnam6502** [comment]
Closed via #277
